Plug-Ins/Menu Editor/MenuItem.swift

```diff
@@ -0,0 +1,207 @@
+    }
+    var markCharacter = "" {
+        didSet {
+            NotificationCenter.default.post(name: MenuItem.markCharacterDidChangeNotification, object: self)
+        }
+    }
+    var styleByte = UInt8(0) {
+        didSet {
+            NotificationCenter.default.post(name: MenuItem.styleByteDidChangeNotification, object: self)
+        }
+    }
+    var menuCommand = UInt32(0) {
+        didSet {
+            NotificationCenter.default.post(name: MenuItem.menuCommandDidChangeNotification, object: self)
+        }
+    }
+    let commandsSize: CommandsSize
+    
+    var isEnabled: Bool = true {
+        didSet {
+            NotificationCenter.default.post(name: MenuItem.enabledDidChangeNotification, object: self)
+        }
+    }
+    
+    var iconImage: NSImage?
+    var iconType: UInt8
+
+    var hasKeyEquivalent: Bool {
+        return !keyEquivalent.isEmpty
+    }
+    
+    var hasCommand: Bool {
+        return commandsSize != .none
+    }
+
+    let manager: RFEditorManager
+    
+    internal init(name: String = "", iconID: Int = Int(0), iconType: UInt8 = 0, keyEquivalent: String = "", markCharacter: String = "", styleByte: UInt8 = UInt8(0), menuCommand: UInt32 = UInt32(0), isEnabled: Bool = true, submenuID: Int = 0, commandsSize: CommandsSize = .none, manager: RFEditorManager) {
+        self.name = name
+        self.iconID = iconID
+        self.iconType = iconType
+        self.keyEquivalent = keyEquivalent
+        self.markCharacter = markCharacter
+        self.styleByte = styleByte
+        self.menuCommand = menuCommand
+        self.isEnabled = isEnabled
+        self.submenuID = submenuID
+        self.commandsSize = commandsSize
+        self.manager = manager
+        
+        super.init()
+        
+        if iconID != 0,
+           let res = manager.findResource(type: ResourceType("ICON"), id: iconID, currentDocumentOnly: false) {
+            res.preview({ img in
+                self.willChangeValue(forKey: "iconImage")
+                self.iconImage = img
+                self.didChangeValue(forKey: "iconImage")
+            })
+        }
+
+    }
+    
+}
+
+extension MenuItem {
+    
+    override func value(forKey key: String) -> Any? {
+        if key == "markCharacter" {
+            return markCharacter
+        } else if key == "keyEquivalent" {
+            return keyEquivalent
+        } else if key == "name" {
+            return name
+        } else if key == "menuID" {
+            return menuID
+        } else if key == "mdefID" {
+            return mdefID
+        } else if key == "menuCommand" {
+            if commandsSize == .int32 {
+                return String(data: Data([UInt8((menuCommand & 0xff000000) >> 24),
+                                          UInt8((menuCommand & 0x00ff0000) >> 16),
+                                          UInt8((menuCommand & 0x0000ff00) >> 8),
+                                          UInt8((menuCommand & 0x000000ff))]), encoding: .macOSRoman)
```

Comment:
As this is a common thing in resources, we actually have an extension to do this (see Extensions.swift in RFSupport):
```suggestion
                return menuCommand.stringValue
```